[BUG] build failure for 8.1.0 on s390x and ppc64le platforms

**Describe the bug**

Valkey 8.1.0 fails to compile on s390x and ppc64le Fedora platforms (https://kojipkgs.fedoraproject.org//work/tasks/5528/131075528/build.log)

```
zmalloc.c:107:37: error: initialization of ‘size_t *’ {aka ‘long unsigned int *’} from incompatible pointer type ‘_Atomic size_t *’ {aka ‘_Atomic long unsigned int *’} [-Wincompatible-pointer-types]
  107 | static size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
      |                                     ^
make[1]: *** [Makefile:538: zmalloc.o] Error 1
```

**To reproduce**

Run 'make' on one of these hardware platforms.

**Expected behavior**

Build completes successfully.

**Additional information**

The following patch appears to resolve it, however I'm unsure whether these platforms should fall into the first cpp branch (without _Atomic) or not?

```
diff -Naurp valkey-8.1.0.orig/src/zmalloc.c valkey-8.1.0/src/zmalloc.c
--- valkey-8.1.0.orig/src/zmalloc.c	2025-04-01 01:56:45.000000000 +1100
+++ valkey-8.1.0/src/zmalloc.c	2025-04-04 11:39:32.762881364 +1100
@@ -101,10 +101,11 @@ static thread_local int thread_index = -
 #if defined(__i386__) || defined(__x86_64__) || defined(__amd64__) || defined(__POWERPC__) || defined(__arm__) || \
     defined(__arm64__)
 static __attribute__((aligned(CACHE_LINE_SIZE))) size_t used_memory_thread_padded[MAX_THREADS_NUM + PADDING_ELEMENT_NUM];
+static size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 #else
 static __attribute__((aligned(CACHE_LINE_SIZE))) _Atomic size_t used_memory_thread_padded[MAX_THREADS_NUM + PADDING_ELEMENT_NUM];
+static _Atomic size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 #endif
-static size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 static atomic_int total_active_threads = 0;
 /* This is a simple protection. It's used only if some modules create a lot of threads. */
 static atomic_size_t used_memory_for_additional_threads = 0;

```

--- a/src/zmalloc.c
+++ b/src/zmalloc.c
@@ -101,10 +101,11 @@ static thread_local int thread_index = -1;
 #if defined(__i386__) || defined(__x86_64__) || defined(__amd64__) || defined(__POWERPC__) || defined(__arm__) || \
     defined(__arm64__)
 static __attribute__((aligned(CACHE_LINE_SIZE))) size_t used_memory_thread_padded[MAX_THREADS_NUM + PADDING_ELEMENT_NUM];
+static size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 #else
 static __attribute__((aligned(CACHE_LINE_SIZE))) _Atomic size_t used_memory_thread_padded[MAX_THREADS_NUM + PADDING_ELEMENT_NUM];
+static _Atomic size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 #endif
-static size_t *used_memory_thread = &used_memory_thread_padded[PADDING_ELEMENT_NUM];
 static atomic_int total_active_threads = 0;
 /* This is a simple protection. It's used only if some modules create a lot of threads. */
 static atomic_size_t used_memory_for_additional_threads = 0;

